lesson20.md

```diff
@@ -0,0 +1,272 @@
+<!-- .slide: id="lesson20" -->
+
+# Basic Frontend - Spring 2020
+
+Lesson 20, Tuesday, 2020-5-19
+
+---
+
+# Homework solution
+
+---
+
+<!-- .slide: id="lesson20:Libraries" -->
+
+# Libraries
+
+---
+
+### Libraries
+
+* Up until now, this was our standard pattern:
+
+```html
+<html>
+  <body>
+    <script src="main.js"></script>
+  </body>
+</html>
+```
+
+All our JavaScript code was in one file called `main.js`
+
+---
+
+### Multiple scripts
+
+* We can add as many scripts as we want:
+
+```html
+<html>
+  <body>
+    <script src="utilities.js"></script>
+    <script src="recipes.js"></script>
+    <script src="main.js"></script>
+  </body>
+</html>
+```
+
+* This allows us to split long scripts into pieces
+
+---
+
+### Multiple scripts
+
+* If we have multiple script tags, we can use all global variables and global functions from all scripts that came _before_ our script.
+* In the example on the last slide, `recipes.js` can use all globals from `utilities.js`, and `main.js` can use all globals from `recipes.js` and `utilities.js`
+
+---
+
+### Scripts in header vs. body
+
+* We put our _own_ scripts at the bottom of the `<body>` tag
+* This allows us to access all HTML elements via DOM API
+* We can put scripts also in the `<head>` tag. These are loaded _before_ the page is rendered:
+
+```html
+<html>
+  <head>
+    <script src="utilities.js"></script>
+  </head>
+  <body>
+    <script src="main.js"></script>
+  </body>
+</html>
+```
+
+---
+
+### Remote scripts
+
+* So far, we loaded scripts only from our local file system
+* However, scripts can also be remote addresses. In the example below, we load the popular `Moment.js` library:
+
+```html
+<html>
+  <head>
+      <script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.25.3/moment.js"></script>
+  </head>
+</html>
+```
+
+---
+
+### moment.js
+
+* `Moment.js` is a popular library for working with date and time
+* It consists of about 5000 lines of JavaScript code
+* It is Open Source (MIT license)
+* When we load `moment.js`, we have access to a single global function called `moment`
+* Documentation: https://momentjs.com/docs/
+
+---
+
+### moment.js example
+
+```html
+<html>
+  <head>
+      <script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.25.3/moment.js"></script>
+      <script>
```

Comment:
To prove that we can use the script loaded in header, isn't better to put the code using moment in the body part. To keep the natural flow going and not to wrongly signal that you can only use the scripts loaded in header also in header.